### PR TITLE
Make parse_time work with individual dt64s and pandas series

### PR DESCRIPTION
**Pull request by dstansby**
- First commit makes individual `datetime64` objects be parsed (previously only arrays of these worked)
- Second commit makes a pandas `Series` of `datetime64` objects be parsed.

### CODE REVIEW

File: sunpy/time/time.py [hunk 1]
```diff
@@ -156,18 +164,20 @@ def parse_time(time_string, time_format='', **kwargs):
     """
     if isinstance(time_string, pandas.Timestamp):
         return time_string.to_pydatetime()
+    elif isinstance(time_string, pandas.Series) and 'datetime64' in str(time_string.dtype):
+        return np.array([dt.to_pydatetime() for dt in time_string])
+    elif isinstance(time_string, pandas.DatetimeIndex):
+        return time_string._mpl_repr()
```
Comment (Cadair) [inline sunpy/time/time.py]: `_mpl_repr`? what's that an array of datetimes? Knowing that pandas loves to be a moving target I would much rather this not be a private method in pandas land.

File: sunpy/time/time.py [hunk 2]
```diff
@@ -126,6 +126,14 @@ def _astropy_time(time):
     return time if isinstance(time, astropy.time.Time) else astropy.time.Time(parse_time(time))
 
 
+def _parse_dt64(dt):
+    """
+    Parse a single numpy datetime64 object
+    """
+    # Validate (in an agnostic way) that we are getting a datetime rather than a date
+    return datetime(*(dt.astype(datetime).timetuple()[:6]))
```
Comment (Cadair) [inline sunpy/time/time.py]: the first 6 elements of this tuple being `year, month, day, hour, min, sec`? what about sub-second numbers?

File: sunpy/time/time.py (hunk 1 above)
Comment (dstansby) [reply sunpy/time/time.py]: No idea, I just moved that `elif` block up from below to sit next to the other pandas blocks.

File: sunpy/time/time.py (hunk 2 above)
Comment (dstansby) [reply sunpy/time/time.py]: Again, I have no idea, this is just copied from the original `datetime64` parsing code here: https://github.com/sunpy/sunpy/pull/2370/files#diff-e4d7773dc71c1ca30c1741901255ba37L170

**Comment by Cadair** [comment]
@dstansby could you add a changelog entry under a 0.9 heading, with the PR number in `[]`. Thanks.

**Pull request merged**